### PR TITLE
bpo-39903: Fix double decref in _elementtree.Element.__getstate__

### PR DESCRIPTION
**Pull request by serhiy-storchaka**
https://bugs.python.org/issue39903

### CODE REVIEW

File: Modules/_elementtree.c [hunk 1]
```diff
@@ -963,33 +963,24 @@ _elementtree_Element___getstate___impl(ElementObject *self)
...
+        if (!attrib) {
+            Py_DECREF(children);
+            return NULL;
+        }
     }
+
+    return Py_BuildValue("{sOsNsNsOsO}",
+                         PICKLED_TAG, self->tag,
+                         PICKLED_CHILDREN, children,
+                         PICKLED_ATTRIB, self->extra->attrib,
```
Comment (ZackerySpytz) [inline Modules/_elementtree.c]: I don't think using `self->extra->attrib` is correct here. I think it should be `attrib`.

**Review by ZackerySpytz — COMMENTED** [review]
It looks like ``test_xml_etree_c`` fails.

File: Modules/_elementtree.c (hunk 1 above)
Comment (serhiy-storchaka) [reply Modules/_elementtree.c]: Indeed, I make an error when manually copied the part of #18735. Thank you.

File: Modules/_elementtree.c [hunk 2]
```diff
@@ -963,33 +963,24 @@ _elementtree_Element___getstate___impl(ElementObject *self)
...
-
-        return NULL;
+        attrib = PyDict_New();
+        if (!attrib) {
+            Py_DECREF(children);
+            return NULL;
+        }
     }
+
+    return Py_BuildValue("{sOsNsNsOsO}",
```
Comment (scoder) [inline Modules/_elementtree.c:978]: Doesn't this leak a reference to `attrib` now? Sorry if this seems to add overhead, I still think it's better for the later maintenance to keep a single call to `Py_BuildValue()`, in favour of a bit more pre/post work.

Comment (scoder) [reply Modules/_elementtree.c:978]: Sorry, my bad. I didn't know the `N` format. Looked it up now, all fine.

**Review by scoder — APPROVED** [review]

**Pull request merged**